### PR TITLE
Add options.dumpException.  Don't re-parse coffee-script - assume we are using source maps.

### PR DESCRIPTION
**Pull request by jwalton**
## Changes in this PR:
- Don't try to read source files that don't exist when injecting source lines.
- Don't recompile coffee-script to get JS source - coffee-script now has source map support, so
  the line numbers should be coffee-script line numbers.  Also, requiring coffee-script 1.6.3 here was wiping
  out our source-map handler from coffee-script 1.7.1.  :P
- Add build script to package.json, so you can `npm run-script build` to build this.
- Make `dumpException` option functional - if false, don't dump the exception to stderr.  If a function, then pass the original exception to the function and let someone else deal with it.
- Remove "MATCH>>" debug logging.

### CODE REVIEW

**Comment by Lendar** [comment]
This PR was helpful to me.

---

I've tested stacktraces of CoffeeScript 1.5 vs 1.6 vs 1.7. 

CoffeeScript output

```
v1.5.0 – at Merger.report (./lib/sprint-report.coffee:151:13)
v1.6.3 – at Merger.report (./lib/sprint-report.coffee:92:11, <js>:149:13)
v1.7.1 – at Merger.report (./lib/sprint-report.coffee:92:11)
```

Express-error v0.0.7 output

```
v1.5.0 – MATCH>> [ '(./lib/sprint-report.coffee:151:13)',
v1.6.3 – Error: ENOENT, no such file or directory './lib/sprint-report.coffee:92:11, <js>'
v1.7.1 – MATCH>> [ '(./lib/sprint-report.coffee:92:11)',
```

**TL;DR express-error is broken for coffee-script v1.6 and v1.7 in different ways**

May be it's good idea to test source maps support.
